src/System.Text.Json/System/Text/Json/JsonReader.cs

```diff
@@ -91,16 +91,97 @@ public JsonReader(ReadOnlySpan<byte> data, SymbolTable symbolTable)
         /// </summary>
         /// <returns>True if the token was read successfully, else false.</returns>
         public bool Read()
+        {
+            if (UseFastUtf8)
+            {
+                return ReadUtf8();
```

Comment:
Would it make sense to split this even more, i.e. have two types: reader for UTF16 and reader for UTF8?

Reply:
Possibly as an API design choice. It won't have significant performance impact, I don't think.